core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BoundedRangeFileInputStream.java

```diff
@@ -62,12 +62,15 @@ public BoundedRangeFileInputStream(FSDataInputStream in, long offset, long lengt
 
   @Override
   public int available() throws IOException {
-    int avail = in.available();
-    if (pos + avail > end) {
-      avail = (int) (end - pos);
-    }
+    final FSDataInputStream inLocal = in;
+    synchronized (inLocal) {
+      int avail = in.available();
```

Comment:
> The only place I found was in the hadoop CompressionInputStream where the returned value is used in the CompressionInputStream.getPos() call which is never actually used.

I was curious about this code, so I took a look at it.  Here is a link if anyone else is interested [CompressionInputStream.java line 88](https://github.com/apache/hadoop/blob/release-2.6.3/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/CompressionInputStream.java#L88)